core/src/main/java/io/parsingdata/metal/expression/derived/GUID.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.expression.derived;
+
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.nio.ByteBuffer.allocate;
+
+import static io.parsingdata.metal.Shorthand.cat;
+import static io.parsingdata.metal.Shorthand.con;
+import static io.parsingdata.metal.Util.checkNotNull;
+
+import java.util.Arrays;
+
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.data.OptionalValueList;
+import io.parsingdata.metal.encoding.Encoding;
+import io.parsingdata.metal.expression.value.ConstantFactory;
+import io.parsingdata.metal.expression.value.ValueExpression;
+
+/**
+ * {@link GUID#guid(String)} creates a ValueExpression to be used as predicate for 16 byte definitions;
+ *
+ * @author Netherlands Forensic Institute.
+ */
+public class GUID {
+    private static final Encoding BIG_ENDIAN = new Encoding();
+
+    /**
+     * Use a String representation of a GUID as predicate.
+     * {@code eq(guid("caa16737-fa36-4d43-b3b6-33f0aa44e76b"))}
+     * Note that the byte order in the encoding matters for the output.
+     * @param guid GUID, for example "caa16737-fa36-4d43-b3b6-33f0aa44e76b"
+     * @return expression to use as predicate
+     */
+    public static ValueExpression guid(final String guid) {
+        final String[] parts = checkNotNull("guid", guid).split("-");
+        if (parts.length != 5) {
+            throw new IllegalArgumentException("Invalid GUID string: " + guid);
+        }
+        return new ValueExpression() {
+
+            @Override
+            public OptionalValueList eval(final Environment environment, final Encoding encoding) {
+                // Note that GUID bytes differ from UUID bytes, as the first 3 parts can be reversed
+                return cat(
+                    cat(
+                        cat(
+                            four(parts[0], encoding),
```

Comment:
Wouldn't depend on the default settings of Encoding, at least not until it is specifically stated :laughing: